packages/core/components/Puck/index.tsx

```diff
@@ -395,12 +398,48 @@ function PuckLayout<
 
   useInjectGlobalCss(iframe.enabled);
 
+  const dispatch = useAppStore((s) => s.dispatch);
   const leftSideBarVisible = useAppStore((s) => s.state.ui.leftSideBarVisible);
   const rightSideBarVisible = useAppStore(
     (s) => s.state.ui.rightSideBarVisible
   );
 
-  const dispatch = useAppStore((s) => s.dispatch);
+  const {
+    width: leftWidth,
+    setWidth: setLeftWidth,
+    sidebarRef: leftSidebarRef,
+    handleResizeEnd: handleLeftSidebarResizeEnd,
+  } = useSidebarResize("left", dispatch);
+
+  const {
+    width: rightWidth,
+    setWidth: setRightWidth,
+    sidebarRef: rightSidebarRef,
+    handleResizeEnd: handleRightSidebarResizeEnd,
+  } = useSidebarResize("right", dispatch);
```

Comment:
These hooks should probably be called within the Sidebar component, along with the refs. There is no reason to keep this state in the main Puck component.

Reply:
The main issue I have with moving the hooks is styling the left and right sidebars. Managing the width gets messy.

Do you have any suggestions for styling them?
I'm mainly concerned about this ternary: ```leftWidth ? `${leftWidth}px` : "var(--puck-side-bar-width)"```

Reply:
Aha I see. This is fine for now.